web/pages/items/[id].vue

```diff
@@ -9,20 +9,42 @@ definePageMeta({
 
 const route = useRoute()
 const activitiesStore = useActivitiesStore()
-const item = computed(() => {
-  if (activitiesStore.items.length === 0) {
-    // TODO: find out how to handle async here
-    activitiesStore.load()
+const item = shallowRef<Item>()
+const error = shallowRef<string>()
+
+const isLoading = shallowRef(false)
+
+onMounted(async () => {
+  if (typeof route.params.id !== 'string') {
+    error.value = 'Invalid item id'
+    return
+  }
+  isLoading.value = true
+  let itemToFind = activitiesStore.items.find(item => item.id === route.params.id)
+  if (!itemToFind) {
+    try {
+      const result = await getItem(route.params.id)
+      itemToFind = transformItem(result)
+    }
+    catch (e) {
+      console.error(e)
+      error.value = 'Item not found'
+    }
+    finally {
+      isLoading.value = false
+    }
   }
```

Comment:
I think there should be another isLoading.value = false somewhere around if statement